Documentation: Data Retention FAQ (#3714)
mattermost/docs: source/administration/data-retention.rst

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Posts and attachments in archived channels are affected by the data retention job. If a post exceeds the age configured for the data retention job it will be deleted from the database.

How long does it take to run a deletion query and does it affect server performance?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Data retention runs the actual deletion query in batches, deleting data in blocks of 1,000 records per query so the database won’t be locked up for extended periods of time with slow, long-running queries (keeping to this limit keeps the query down to a few milliseconds execution time on the database itself).

Reviewer: ```suggestion
Data retention runs the actual deletion query in batches, deleting data in blocks of 1 000 records per query. This is so the database won’t be locked up for extended periods of time with long-running queries. Keeping to this limit keeps the query down to a few milliseconds' execution time on the database itself.
```

Author: This looks good

> Each batch of data is deleted based on indexes, so the queries are generally quite quick to execute on such small batches meaning the server will stay fully responsive while the process is running.

Reviewer: ```suggestion
Each batch of data is deleted based on indexes, meaning the queries are generally quick to execute on such small batches. This helps the server remain fully responsive while the process is running.
```

Author: Each batch of data is deleted based on indexes making the queries quick to execute on small batches. This helps the server remain fully responsive while the process is running

@justinegeffen here is my suggestion what do you think?

Reviewer: I really like it, thanks. I've added a new suggestion with your change. :)

> Each batch of data is deleted based on indexes, so the queries are generally quite quick to execute on such small batches meaning the server will stay fully responsive while the process is running.

Reviewer: ```suggestion
Each batch of data is deleted based on indexes making the queries quick to execute on small batches. This helps the server remain fully responsive while the process is running.
```

How do I know whether the data retention job is running/scheduled?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> When the job scheduler identifies that it is time for the data retention job to be scheduled again for immediate running (i.e. at the time they have configured for it to run each day), a debug level logline is printed: ``Scheduling data retention job``.

Reviewer: ```suggestion
The job scheduler runs the data retention job based on the time specified in the configuration settings. At this time a ``DEBUG``-level log line is printed: ``Scheduling data retention job``.
```

> When a job server then picks up that scheduled job in order to execute it there will be a debug log as follows: ``Worker EnterpriseDataRetention: Received a new candidate job``.

Reviewer: ```suggestion
When a job server picks up that scheduled job in order to execute it, a ``DEBUG``-level log line is generated: ``Worker EnterpriseDataRetention: Received a new candidate job``.
```

Author: When a job server picks up that scheduled job for execution, a ``DEBUG`` -level log line is generated: ``Worker EnterpriseDataRetention: Received a new candidate job``.

@justinegeffen my second suggestion

> When the job is complete, an ``INFO``-level log line ``Worker EnterpriseDataRetention: Job is complete`` is generated. 

Reviewer: ```suggestion
When the job is complete, an ``INFO``-level log line is generated: ``Worker EnterpriseDataRetention: Job is complete``. 
```

> When the job is complete there is an info level logline: ``Worker EnterpriseDataRetention: Job is complete``.

Reviewer: ```suggestion
When the job is complete, an ``INFO``-level log line ``Worker EnterpriseDataRetention: Job is complete`` is generated. 
```
